src/jit/lower.cpp

```diff
@@ -979,16 +1048,32 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
                 GenTreeFieldList* fieldListPtr = arg->AsFieldList();
                 assert(fieldListPtr->IsFieldListHead());
 
+                // There could be up to 2-4 PUTARG_REGs in the list (3 or 4 can only occur for HFAs)
+                regNumber argReg = info->regNum;
                 for (unsigned ctr = 0; fieldListPtr != nullptr; fieldListPtr = fieldListPtr->Rest(), ctr++)
                 {
                     GenTreePtr curOp  = fieldListPtr->gtOp.gtOp1;
                     var_types  curTyp = curOp->TypeGet();
 
                     // Create a new GT_PUTARG_REG node with op1
                     GenTreePtr newOper = comp->gtNewPutArgReg(curTyp, curOp);
+                    newOper->gtRegNum  = argReg;
```

Comment:
I wonder if it would be best to add a `regNum` parameter to `gtNewPutArgReg` and set `gtRegNum` in that function rather than doing so afterwards.